.github/workflows/analysis-of-endpoint-connections.yml

```diff
@@ -23,7 +23,16 @@ jobs:
         run: |
           git diff --name-only origin/${{ github.event.pull_request.base.ref }} HEAD > modified_files.txt
 
-      - name: Display modified files
+      - name: Set up JDK 21
+        uses: actions/setup-java@v2
+        with:
+          java-version: '21'
+          distribution: 'adopt'
+
+      - name: Build analysis-of-endpoint-connections
         run: |
-          echo "Modified files in this pull request:"
-          cat modified_files.txt
+          ./gradlew :supporting_scripts:analysis-of-endpoint-connections:build --warning-mode all
+          ./gradlew :supporting_scripts:analysis-of-endpoint-connections:shadowJar
+
+      - name: run analysis-of-endpoint-connections
+        run: java -jar supporting_scripts/analysis-of-endpoint-connections/build/libs/analysis-of-endpoint-connections-1.0-SNAPSHOT-all.jar $(cat modified_files.txt)
```

Comment:
I feel like the explicit path of the analysis script (`supporting_scripts/analysis-of-endpoint-connections/build/libs/analysis-of-endpoint-connections-1.0-SNAPSHOT-all.jar`) could be a potential source for "unexplained errors".
If possible, I´d love to see you use a return value of some kind from the Gradle Build steps that link to the final JAR file or define it from the beginning. This way, changes in the `settings.gradle` file (like `group` or `version`) would not also require changes here. 

After some initial googling from my side, I did not find a fitting parameter for Gradle though - maybe you can take another look at it?

Reply:
I looked into the matter and did not yet come up with a satisfactory solution. I would like to move this into a follow up PR.